src/engine/source/kvdb/src/kvdb.cpp

```diff
@@ -1,561 +1,509 @@
         return false;
     }
 
-    if(column_family_name.empty()) {
-        LOG(ERROR) << "[" << __func__ << "]" << " can't write to a Transaction to a family column with no name." << std::endl;
+    rocksdb::Transaction *txn = m_txndb->BeginTransaction(kOptions.write);
+    if (!txn)
+    {
+        WAZUH_LOG_ERROR("couldn't begin in transaction");
         return false;
     }
 
-    UpdateColumnFamiliesList();
-
-    int i = CFIndexInAvailableArray(column_family_name);
-    if(i) {
-        s = TransactionDB::Open(DBOptions(), txn_db_options, kKvDbPath, column_families, &handlers, &txn_db);
-        if(s.ok()) {
-            Transaction* txn = txn_db->BeginTransaction(write_options);
-            if(txn) {
-                for(auto handle : handlers) {
-                    // find the correct CF handle to be erased
-                    if(!column_family_name.compare(handle->GetName())) {
-                        for(auto pair : pairsVector) {
-                            std::string const key = pair.first;
-                            std::string const value = pair.second;
-                            if(key.empty()) {
-                                LOG(ERROR) << "[" << __func__ << "]" << " can't write to a Transaction to a family column with no key." << std::endl;
-                                return false;
-                            }
-                            // Write a key in this transaction
-                            s = txn->Put(handle, key, value);
-                            if(s.ok()) {
-                                continue;
-                            }
-                            else {
-                                LOG(ERROR) << "[" << __func__ << "]" << " couldn't execute Put in transaction -breaking loop-, error: " << s.code() << std::endl;
-                                result =  false;
-                            }
-                        }
-                        s = txn->Commit();
-                        if (s.ok()) {
-                            LOG(ERROR) << "[" << __func__ << "]" << " transaction commited OK" << std::endl;
-                            delete txn;
-                            result =  true;
-                        }
-                        else {
-                            LOG(ERROR) << "[" << __func__ << "]" << " couldn't commit transaction, error: " << s.code() << std::endl;
-                            result =  false;
-                        }
-                    }
-                }
-                handlers.clear();
-            }
-            else {
-                LOG(ERROR) << "[" << __func__ << "]" << " couldn't begin transaction, error: " << s.code() << std::endl;
-                result =  false;
-            }
+    for (auto & [key, value] : pairsVector)
+    {
+        if (key.empty())
+        {
+            WAZUH_LOG_ERROR("can't write to a Transaction to a family "
+                            "column with no key.");
+            return false;
```

Comment:
returning here is a memory leak

Reply:
this should be fixed [here](https://github.com/wazuh/wazuh/issues/12935)